spec.bs

```diff
@@ -675,28 +697,59 @@ To <dfn>score and rank a bid</dfn> given an [=auction config=] |auctionConfig|,
+1. Otherwise if |score| is greater than |leadingBidInfo|'s [=leading bid info/top score=]:
   1. Set |leadingBidInfo|'s [=leading bid info/top bids count=] to 1.
-1. Otherwise if |score| equals to |leadingBidInfo|'s [=leading bid info/top score=]:
+  1. Set |updateLeadingBid| to true.
+1. Otherwise if |score| equals |leadingBidInfo|'s [=leading bid info/top score=]:
   1. Increment |leadingBidInfo|'s [=leading bid info/top bids count=] by 1.
-  1. Set |leadingBidInfo|'s [=leading bid info/leading bid=] to |maybeModifiedBid| with 1 in
-    |leadingBidInfo|'s [=leading bid info/top bids count=] chance.
+  1. Set |updateLeadingBid| to true with 1 in |leadingBidInfo|'s [=leading bid info/top bids count=]
+    chance.
+  1. If |updateLeadingBid| is false:
+    1. [=Update highest scoring other bid=] with |score|, |leadingBidInfo|'s
+      [=leading bid info/leading bid=], and |leadingBidInfo|.
+1. Otherwise if |score| is greater than or equal to |leadingBidInfo|'s
+  [=leading bid info/second highest score=]:
+  1. [=Update highest scoring other bid=] with |score|, |bidValue|, and |leadingBidInfo|.
+1. If |updateLeadingBid| is true:
+  1. [=Update highest scoring other bid=] with |leadingBidInfo|'s [=leading bid info/top score=],
+    |leadingBidInfo|'s [=leading bid info/leading bid=], and |leadingBidInfo|.
+  1. Set |leadingBidInfo|'s [=leading bid info/top score=] to |score|.
+  1. Set |leadingBidInfo|'s [=leading bid info/leading bid=] to |generatedBid|.
+  1. Set |leadingBidInfo|'s [=leading bid info/auction config=] to |auctionConfig|.
+  1. Set |leadingBidInfo|'s [=leading bid info/bidding data version=] to |biddingDataVersion|.
+  1. TODO: Set |leadingBidInfo|'s [=leading bid info/scoring data version=].
+
+</div>
 
+<div algorithm>
+To <dfn>update highest scoring other bid</dfn> given a {{double}} |score|, a
+[=generated bid=]-or-null |bid|, and a [=leading bid info=] |leadingBidInfo|:
+
+  1. If |bid| is null, return.
+  1. If |score| is greater than |leadingBidInfo|'s [=leading bid info/second highest score=]:
+    1. Set |leadingBidInfo|'s [=leading bid info/highest scoring other bid=] to |bid|.
+    1. Set |leadingBidInfo|'s [=leading bid info/highest scoring other bids count=] to 1.
+    1. Set |leadingBidInfo|'s [=leading bid info/second highest score=] to score.
+  1. Otherwise if |score| is equal to |leadingBidInfo|'s [=leading bid info/second highest score=]:
+    1. Increment |leadingBidInfo|'s [=leading bid info/highest scoring other bids count=] by 1.
+    1. Set |leadingBidInfo|'s [=leading bid info/highest scoring other bid=] to |bid| with 1 in
+      |leadingBidInfo|'s [=leading bid info/highest scoring other bids count=] chance.
 </div>
 
-To <dfn>create a request</dfn> given a [=URL=] |url|, a [=string=] <dfn for="create a request">
-|accept|</dfn>, and an [=origin=] or null <dfn for="create a request">|origin|</dfn>:
+To <dfn>create a request</dfn> given a [=URL=] |url|, a [=string=] |accept|, and an [=origin=] or

```

Comment:
I'm not sure what's going on, but for some reason, when I click on url, accept, or origin in the rendered HTML, it doesn't highlight references like it does for other variables.

Not sure why it's behaving differently in this algorithm?